spec/knapsack_pro/adapters/rspec_adapter_spec.rb

```diff
@@ -57,19 +57,36 @@
 
     subject { described_class.test_file_cases_for(slow_test_files) }
 
-    before do
-      logger = instance_double(Logger)
-      expect(KnapsackPro).to receive(:logger).and_return(logger)
-      expect(logger).to receive(:info).with("Generating RSpec test examples JSON report for slow test files to prepare it to be split by test examples (by individual test cases). Thanks to that, a single slow test file can be split across parallel CI nodes. Analyzing 5 slow test files.")
+    context 'when the rake task to detect RSpec test examples succeeded' do
+      it 'returns test example paths for slow test files' do
+        logger = instance_double(Logger)
+        allow(KnapsackPro).to receive(:logger).and_return(logger)
+        allow(logger).to receive(:info)
+
+        cmd = 'bundle exec rake knapsack_pro:rspec_test_example_detector'
+        env = { 'RACK_ENV' => 'test', 'RAILS_ENV' => 'test' }
+        expect(Kernel).to receive(:system).with(env, cmd).and_return(true)
+
+        rspec_test_example_detector = instance_double(KnapsackPro::TestCaseDetectors::RSpecTestExampleDetector)
+        expect(KnapsackPro::TestCaseDetectors::RSpecTestExampleDetector).to receive(:new).and_return(rspec_test_example_detector)
+
+        test_file_example_paths = double
+        expect(rspec_test_example_detector).to receive(:test_file_example_paths).and_return(test_file_example_paths)
+
+        expect(subject).to eq test_file_example_paths
 
-      cmd = 'RACK_ENV=test RAILS_ENV=test bundle exec rake knapsack_pro:rspec_test_example_detector'
-      expect(Kernel).to receive(:system).with(cmd).and_return(cmd_result)
+        expect(logger).to have_received(:info).with("Generating RSpec test examples JSON report for slow test files to prepare it to be split by test examples (by individual test cases). Thanks to that, a single slow test file can be split across parallel CI nodes. Analyzing 5 slow test files.")
+      end
```

Comment:
Note for the future:

These tests need too many doubles so it's pretty fragile.

If we want to test behavior with so many side effects, let's use an integration test. If we want to test specifics (`Kernel` receives some arguments), let's create a separate method and unit test just that (without stubbing so many things).

I would have picked the latter in this case.